mmv1/third_party/terraform/resources/resource_storage_transfer_job.go

```diff
@@ -370,6 +379,44 @@ func httpDataSchema() *schema.Resource {
 	}
 }
 
+func azureBlobStorageDataSchema() *schema.Resource {
+	return &schema.Resource{
+		Schema: map[string]*schema.Schema{
+			"storage_account": {
+				Required:    true,
+				Type:        schema.TypeString,
+				Description: `The name of the Azure Storage account.`,
+			},
+			"container": {
+				Required:    true,
+				Type:        schema.TypeString,
+				Description: `The container to transfer from the Azure Storage account.`,
+			},
+			"path": {
+				Required:    true,
+				Type:        schema.TypeString,
+				Description: `Root path to transfer objects. Must be an empty string or full path name that ends with a '/'. This field is treated as an object prefix. As such, it should generally not begin with a '/'.`,
+			},
+			"azure_credentials": {
```

Comment:
It looks like this is input-only and should possibly be ForceNew: https://cloud.google.com/storage-transfer/docs/reference/rest/v1/TransferSpec#AzureBlobStorageData

Reply:
I'm not sure, the AWS credentials aren't ForceNew either. I'm assuming it behaves the same way.

Reply:
hmm... the docs also say that AWS creds are input-only. I think that's probably a bug waiting to happen; we should trust the docs and make this ForceNew.

Reply:
The documentation is stale. I got hold of an Azure account for testing and updating the SAS Token doesn't recreate the resource.

Unfortunately, Storage Transfer doesn't log anything in Stackdriver and I created it in the console.


```
C02S62ZFFVH6:Git mahamed$ curl -H "Authorization: Bearer $(gcloud auth application-default print-access-token)" https://storagetransfer.googleapis.com/v1/transferJobs/17436582952808956420?project_id=550924169191
{
  "name": "transferJobs/17436582952808956420",
  "description": "Provider Development",
  "projectId": "REDACTED",
  "transferSpec": {
    "gcsDataSink": {
      "bucketName": "videndum-tmp"
    },
    "objectConditions": {},
    "transferOptions": {},
    "azureBlobStorageDataSource": {
      "storageAccount": "mahamedgcstransfer",
      "container": "dev"
    }
  },
  "schedule": {
    "scheduleStartDate": {
      "year": 2021,
      "month": 5,
      "day": 19
    },
    "scheduleEndDate": {
      "year": 2021,
      "month": 5,
      "day": 19
    },
    "endTimeOfDay": {}
  },
  "status": "ENABLED",
  "creationTime": "2021-05-19T14:29:58.948051753Z",
  "lastModificationTime": "2021-05-19T14:50:30.018660834Z",
  "latestOperationName": "transferOperations/transferJobs-17436582952808956420--7984150713314667712"
}
```

Reply:
![image](https://user-images.githubusercontent.com/25100905/118835111-9d743a80-b8ba-11eb-8af9-d25af5d1a7ca.png)

Reply:
okay great, thanks for testing this! Sorry for the delayed response; just saw the notification.